drivers/azure/azureutil/clients.go

```diff
@@ -49,6 +52,7 @@ func (a AzureClient) resourceGroupsClient() resources.GroupsClient {
 	c.Client.UserAgent += fmt.Sprintf(";docker-machine/%s", version.Version)
 	c.RequestInspector = withInspection()
```

Comment:
@nathanleclaire  that could actually work, though as my note indicates, we're planning to make this stuff part of the SDK sometime soon via a factory pattern or something. So things will get better. I may actually submit a separate PR for that.